server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java

```diff
@@ -1645,4 +1650,39 @@ private static <T> Set<T> collectionToSet(Collection<T> collection) {
                 .map(HashSet::new)
                 .orElse(null);
     }
+
+    private static void updateOrganizationBroker(RealmModel realm, IdentityProviderRepresentation representation, KeycloakSession session) {
+        if (!Profile.isFeatureEnabled(Feature.ORGANIZATION)) {
+            return;
+        }
+
+        IdentityProviderModel existing = realm.getIdentityProvidersStream()
+                .filter(p -> Objects.equals(p.getAlias(), representation.getAlias()) || Objects.equals(p.getInternalId(), representation.getInternalId()))
+                .findFirst().orElse(null);
+
+        if (existing == null) {
+            return;
+        }
```

Comment:
Why return here? If you are importing an IDP, don't you want to check if the representation matches an org (potentially) and one of its domain regardless of it existing already?